lib/db_connection.ex

```diff
@@ -980,6 +1192,17 @@ defmodule DBConnection do
     end
   end
 
+  defp checkin(%DBConnection{} = conn, fun, meter, opts) do
+    with {:ok, conn_state, meter} <- fetch_info(conn, meter) do
+      return = fun.(conn, conn_state, meter, opts)
+      checkin(conn, opts)
+      return
+    end
+  end
+  defp checkin(pool, fun, meter, opts) do
+    run(pool, fun, meter, opts)
```

Comment:
This is for API completeness. Also someone using the ownership pool may wish to commit/rollback their transaction before returning connection to the inner pool. I am happy to remove this option.